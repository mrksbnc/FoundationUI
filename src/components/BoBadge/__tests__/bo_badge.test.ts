import {
	BoBadge,
	BoBadgeSize,
	BoBadgeType,
	BoBadgeVariant,
} from '@/components/BoBadge';
import { mount } from '@vue/test-utils';
import { describe, expect, it } from 'vitest';

describe('BoBadge', () => {
	it('should match the snapshot', () => {
		const wrapper = mount(BoBadge, {
			props: {
				label: 'Badge',
			},
		});

		expect(wrapper.html()).toMatchSnapshot();
	});

	it('should render the correct label', () => {
		const label = 'Badge';
		const wrapper = mount(BoBadge, {
			props: {
				label,
			},
		});

		expect(wrapper.text()).toBe(label);
	});

	it('should render the correct variant', () => {
		const wrapper = mount(BoBadge, {
			props: {
				label: 'Badge',
				variant: BoBadgeVariant.green,
			},
		});

		expect(wrapper.classes()).toContain('bg-green-700');
	});

	it('should render the correct size', () => {
		const wrapper = mount(BoBadge, {
			props: {
				label: 'Badge',
				size: BoBadgeSize.large,
			},
		});

		console.log(wrapper.html());
		expect(wrapper.classes()).toContain('py-1.5');
	});

	it('should render the correct type', () => {
		const wrapper = mount(BoBadge, {
			props: {
				label: 'Badge',
				type: BoBadgeType.outline,
			},
		});

		expect(wrapper.classes()).toContain('border-blue-700');
	});
});
