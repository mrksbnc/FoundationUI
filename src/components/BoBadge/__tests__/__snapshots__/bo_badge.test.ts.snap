// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BoBadge > should match the snapshot 1`] = `"<span class="bo-badge flex max-w-fit justify-center items-center content-center space-x-1 rounded-lg shadow-md cursor-default select-none badge-default px-2.5 py-1 text-base bg-gray-100 text-gray-600"><!--v-if--><p class="bo-badge__label font-medium">Badge</p><!--v-if--></span>"`;
